test/conformance/ingress/class.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"knative.dev/networking/pkg/apis/networking"
+	"knative.dev/networking/pkg/apis/networking/v1alpha1"
+	"knative.dev/networking/test"
+	"knative.dev/pkg/reconciler"
+)
+
+// TestIngressClass verifies that kingrress does not picck ingress up when ingress.class annotation is incorrect.
+func TestIngressClass(t *testing.T) {
+	t.Parallel()
+	ctx, clients := context.Background(), test.Setup(t)
+
+	// Create a backend service to create valid ingress except for invalid ingress.class.
+	name, port, _ := CreateRuntimeService(ctx, t, clients, networking.ServicePortNameHTTP1)
+	ingressBackend := &v1alpha1.IngressBackend{
+		ServiceName:      name,
+		ServiceNamespace: test.ServingNamespace,
+		ServicePort:      intstr.FromInt(port),
+	}
+
+	tests := []struct {
+		name        string
+		annotations map[string]string
+	}{{
+		name: "nil",
+	}, {
+		name:        "omitted",
+		annotations: map[string]string{},
+	}, {
+		name:        "incorrect",
+		annotations: map[string]string{networking.IngressClassAnnotationKey: "incorrect"},
+	}, {
+		name:        "empty",
+		annotations: map[string]string{networking.IngressClassAnnotationKey: ""},
+	}}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			createIngressWithAnnotations(ctx, t, clients, test.annotations, ingressBackend)
```

Comment:
I would rename the function name as `verifyIngressWithAnnotations`. WDYT?

Reply:
Yes, sounds good. Thank you. Updated.